src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2817,6 +2817,19 @@ bool SessionImpl::addTorrent_impl(const TorrentDescriptor &source, const AddTorr
                 loadTorrentParams.name = contentName;
         }
 
+        const auto nativeIndexes = torrentInfo.nativeIndexes();
+
+        Q_ASSERT(p.file_priorities.empty());
+        Q_ASSERT(addTorrentParams.filePriorities.isEmpty() || (addTorrentParams.filePriorities.size() == nativeIndexes.size()));
+        QList<DownloadPriority> filePriorities = addTorrentParams.filePriorities;
+
+        // Filename filter should be applied before `findIncompleteFiles()` is called.
```

Comment:
In fact, this is not required, since `findIncompleteFiles()` is executed asynchronously. But it's still more logical to have them in that order.